### ISSUE TITLE
logind does not recognize ID 17ef:3060 Lenovo ThinkPad Dock 

### ISSUE THREAD

**Issue by xypron** [systemd/systemd#23684]
lsusb identifies my docking station as: ID 17ef:3060 Lenovo ThinkPad Dock

But when booting with the lid closed the laptop suspends before showing the login prompt with these settings;

/etc/systemd/logind.conf
[Login]
#HandleLidSwitch=suspend
#HandleLidSwitchExternalPower=suspend
#HandleLidSwitchDocked=ignore

The following is a workaround:

/etc/systemd/logind.conf.d/10-lid.conf
[Login]
HandleLidSwitchExternalPower=ignore

But I think discovering the USB device ID should be enough to stop suspending the laptop.

Currently logind relies on SW_DOCK events which the kernel emits based on ACPI. Not all USB docking stations come from the same vendor as the laptops. So an ACPI approach is bound to fail.

Either we need a new kernel driver that generates SW_DOCK events or we could use a udev rules or logind would have to look up the USB devices.

Of course multiple USB docking stations could be connected to a laptop at the same time. Only when the last one is disconnected we are undocked. This excludes a udev rules based approach as well as an SW_DOCK based solution.

This leaves us with logind having to lookup up the USB devices in a database.

Distro: Ubuntu Kinetic with systemd-249.11

Cf. closed #18333

Best regards

Heinrich

**Comment by keszybz** [comment]
See also [16045](https://github.com/systemd/systemd/issues/16045).

We only suspend by default when `manager_is_docked_or_external_displays()` returns true.
Do you have an external monitor connected to the dock?
If yes, then we have a bug in `manager_count_external_displays()`. Fixing that would probably be easier than building a database of docking stations.

**Comment by xypron** [comment]
Yes, I have an external connected to the docking station via its DisplayPort.
